src/Security/Security.php

```diff
@@ -303,12 +392,8 @@ public static function permissionFailure($controller = null, $messageSet = null)
                 $message = $messageSet['default'];
             }
 
-            // Somewhat hackish way to render a login form with an error message.
-            $me = new Security();
-            $form = $me->LoginForm();
-            $form->sessionMessage($message, ValidationResult::TYPE_WARNING);
-            Session::set('MemberLoginForm.force_message', 1);
-            $loginResponse = $me->login();
+            Security::setLoginMessage($message, ValidationResult::TYPE_WARNING);
+            $loginResponse = (new Security())->login(new HTTPRequest('GET', '/'));
```

Comment:
OK, I wanted to understand more why we need to do this rather than advocate increasing the scope of the PR.